src/DwitTech.AccountService.Core/Services/UserService.cs

```diff
@@ -357,5 +348,53 @@ public async Task<bool> ResetPassword(string userEmail)
                 throw;
             }
         }
+
+        private async Task<bool> SetPassword(string newPassword, string userEmail)
+        {
+            var newPasswordHash = StringUtil.HashString(newPassword);
+            var user = await _userRepository.GetUserByEmail(userEmail);
+            await _userRepository.UpdateUserLoginAsync(user, newPasswordHash);
+            _logger.LogInformation(1, $"Password for the user with ID {user.Id} was changed successfully");
+            return true;
+        }
+
+        public async Task<bool> HandlePasswordReset(string passwordResetToken, PasswordResetModel passwordResetModel)
+        {
+
+            try
+            {
+                var getUserValidationCode = await _userRepository.GetUserValidationCode(passwordResetToken, CodeType.ResetToken);
+                var resetPasswordResult = await UpdatePassword(getUserValidationCode.UserId, passwordResetModel);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("User does not exist", ex);
+                throw new NullReferenceException("User with the given token does not exist");
+            }
+        }
+
+        public async Task<bool> UpdatePassword(int userId, PasswordResetModel passwordResetModel)
+        {
+            var getUserLoginsModel = await _userRepository.GetUserLoginsByUserId(userId);
```

Comment:
Variable name should not connote an action. I believe removing get should solve it.